src/durable-cache/src/lib.rs

```diff
@@ -174,57 +177,80 @@ impl<C: DurableCacheCodec> DurableCache<C> {
         }
     }
 
+    /// Return all entries stored in the cache, without syncing with the durable store.
+    pub fn entries_local(&self) -> impl Iterator<Item = (&C::Key, &C::Val)> {
+        self.local.iter()
+    }
+
     /// Durably set `key` to `value`. A `value` of `None` deletes the entry from the cache.
+    ///
+    /// Failures will update the cache and retry until the cache is written successfully.
     pub async fn set(&mut self, key: &C::Key, value: Option<&C::Val>) {
-        self.set_many(&[(key, value)]).await
+        while self.try_set(key, value).await.is_err() {}
```

Comment:
Silently retrying errors can make things hard to debug in production. Depending on how expected it is to get an error, we should log and/or have a metric.

Reply:
Oh I see you've just moved the retry logic in this PR. In that case, I'm fine merging without doing this, but let's make sure to circle back and add it when we get this hooked up to the expression cache impl